### ISSUE TITLE
Material Design 3 (Material You) support

### ISSUE THREAD

**Issue by ThatOneCalculator** [SchildiChat/SchildiChat-android#137]
Upgrading to Material Design 3 might help the app feel a bit more modern and stand out even more from Element.

**Comment by SpiritCroc** [comment]
I agree that this would make the app feel more native for recent Android versions.
However, last time I tried out Android 12, I hated the design and downgraded to Android 11.
So I'd prefer it to be a user-configurable setting (probably best just as another theme selection in our current theme picker), and don't count on me to implement it.

**Comment by ThatOneCalculator** [comment]
Fair enough. I feel like *when done right* Material You can look really nice, but then again I'm a sucker for uniform design.

**Comment by ThatOneCalculator** [comment]
The problem with making it a theme is it'd require a redesign of the bottom/top bar, buttons, etc

**Comment by SpiritCroc** [comment]
Yeah, uniform design is nice. I was just thinking to myself some minutes ago that I might be able to really like Material You, if I would find the time to tweak it ROM-side to make everything look like Material Design 1.0 again.

I don't see the redesign issue, isn't that controlled by themes either way? At least for system components it should be. Are there any Schildi-specific UI elements that one would need to change?

**Comment by SpiritCroc** [comment]
Relevant resource: https://material.io/blog/migrating-material-3

But well, I'm already doing more investigation than I wanted for not wanting to implement this myself, so I'll stop thinking about it for now :eyes:

**Comment by yavko** [comment]
Would love and switch to SchildiChat if this was supported!

**Comment by anoosa1** [comment]
What about the material you icons atleast?

**Comment by SpiritCroc** [comment]
> What about the material you icons atleast?

Also open for contributions - but should resemble our default icon. We have SVGs in https://github.com/SchildiChat/schildichat-meta if anybody wants to have a try

**Comment by NathanC** [comment]
@SpiritCroc I took a stab at it. I'm not a graphic designer by any stretch, but I'm happy with how it turned out—it would be great to have a themed icon, for consistency with most of my apps (especially the ones I choose to keep on my homescreen, like SchildiChat).

The svg can be found [here](https://github.com/NathanC/Icons/blob/main/ic_launcher_monochrome_sc.svg)
![ic_launcher_monochrome_sc](https://user-images.githubusercontent.com/6182419/204407653-0f4cf6b0-26da-416c-8161-fefd262a6735.svg)

I had a bit of trouble building SchildiChat, so I just inserted this as `<monochrome>` within `anydpi-v26` within my own app (**note: you need to set it in both the normal and round xmls**) and deployed it next to SchildiChat's existing icon on my homescreen:

![example_themed_sc_icon](https://user-images.githubusercontent.com/6182419/204408079-d3466f17-4c34-4194-a99f-a543ead7130c.png)

----

I played around with a few designs. I think you either need to make the shell filled and the feet/horns outlined, or vice versa, and I think this looks better—more like a turtle, and more balanced.

I played around with the overall sizing to try and get it within the recommended safe area for adaptive/themeable icons, and ended up compressing it vertically a bit, otherwise it was either too small overall or portions were too close to the border (see the general sizing of other themed icons I have in the screenshot).

I went through a lot of iterations with the internal spacing between the shell portions and stroke width for the foot/horn, and I think wider strokes looks better.

Let me know if you want me to make any changes :)

**Comment by yavko** [comment]
Shouldn't this be as a PR?

**Comment by NathanC** [comment]
Probably, yeah, just wanted to get initial feedback from this thread. I'll open a PR in a day or two.

**Comment by yavko** [comment]
👍

**Comment by SpiritCroc** [comment]
@NathanC thanks, I like it! I'd be interested in seeing the variant with the shell outlined and the feet filled in a screenshot - for a short moment, I thought the feet looked like glasses in your current design.

I doubt we'd need to set it as round icon for SchildiChat, since we currently do not have any specific round icon set, which it would need to override. I figured that the amount of people requiring a round icon but not being able to generate one using adaptive icons is probably quite small (only users of Google's Android 7.1, I think?), so I considered it not worth the effort to maintain.
